lfs.c

```diff
@@ -4911,3 +4911,124 @@ int lfs_migrate(lfs_t *lfs, const struct lfs_config *cfg) {
 }
 
 #endif
+
+#if LFS_THREAD_SAFE
+
+#define CREATE_LFS_TS_1(ret, function, b_type, b) \
+    ret _ts ## function(b_type b)                 \
+    {                                             \
+        int err = lfs->cfg->lock(lfs->cfg);       \
```

Comment:
This is a good way to implement this. A future feature I want to look at is optionally collapsing configuration at compile time, but that will be something for the future.